src/generators/javascript/renderers/ClassRenderer.ts

```diff
@@ -36,6 +36,12 @@ ${this.indent(this.renderBlock(content, 2))}
       content.push(this.renderBlock([getter, setter]));
     }
 
+    if (this.model.additionalProperties !== undefined && this.model.additionalProperties instanceof CommonModel) {
+      const getter = 'getAdditionalProperty(key){ return _additionalProperties[key]; }';
+      const setter = 'setAdditionalProperty(key, value) { _additionalProperties[key] = value; }';
+      content.push(this.renderBlock([getter, setter]));
+    }
+
```

Comment:
it should be rendered inside `additionalContent` preset:

```ts
{
  class: {
    additionalContent({ renderer, model, content }) {
      if (!(this.model.additionalProperties instanceof CommonModel)) return content;
    
      const getter = 'getAdditionalProperty(key){ return _additionalProperties[key]; }';
      const setter = 'setAdditionalProperty(key, value) { _additionalProperties[key] = value; }';
      this.renderBlock([content, getter, setter]);
     }
  }
}
```

Reply:
I think you mean the extra functions that are not really encapsulations right? 🤔

Reply:
Yes, for that `additionalContent` exists - for extra function. Encapsulation is a problem, but you can create separate presets to make your encapsulation

```ts
new TypeScriptGenerator({ presets: [ADDITIONALPROPETIES, PATTERN_PROPERTES] })
```

Reply:
We can make separate presets for everything, argument, type signature etc. but then it would be hard to use.